### ISSUE TITLE
Awesome script, I love it, just one question

### ISSUE THREAD

**Issue by bluefiberbread** [ccloli/E-Hentai-Downloader#92]
What does "Failed! (Wrong MIME)" mean?

**Comment by ewasion** [comment]
MIME refers to [MIME types](https://en.wikipedia.org/wiki/MIME), used to tell the browser what type a file should be interpreted as.

**Comment by ccloli** [comment]
Hi, that means the file type returned by server is wrong, it should be an image file (`image/*`) but exactly it's another file.

Please open any pages to check if you can see the image file, and click the link `Download original image ......` below (if you cannot find it, turn to another page) to check if you can get the original image. 

If you cannot get the image, please following the steps below:

1. If it says your image limits is exceed, or your IP is blocked (#84), please wait the time it says until you can access to it;
2. If it says you are not login, or directs you to login page, please login again, also make sure you've enabled cookies and third-party cookies (if you're visiting exhentai, the cookies on e-hentai maybe blocked by browser, #86);
3. If you're using a proxy server, maybe the proxy server blocks the file and gives you the warning, it may also happens if your DNS, firewall, your ISP or [your country](https://ehwiki.org/wiki/Technical_Issues#E-Hentai_Galleries) (like South Korea) has the rules to block adult websites.

If everything is fine, when you get the error, press F12, select Console tab, then copy and paste the logs that prefixed with `[EHD]` to here.

**Comment by bluefiberbread** [comment]
I think it's an issue with the servers and not with the script itself. I've tried another script and it's also producing the same errors. Are you able to download galleries in original resolution just fine? Might just be my connection then... a small suggestion, could you make it download a blank image if it fails with the page number as the file name?

**Comment by ccloli** [comment]
@bluefiberbread 
Please try going to https://exhentai.org/s/7b87643838/530350-1 , then click the link `Download original 700 x 979 563.8 KB source` at the bottom of the page. It should gives you the image file, but if it's something else, check the steps from my previous comment. If nothing matched, let me know what you see on that page.

**Comment by bluefiberbread** [comment]
Yes, it works fine to do it manually. Can you download a full gallery in its original size without any errors right now?

**Comment by ccloli** [comment]
It works fine here. That's strange, if you still have the problem, can you help me to:

1. Press F12, switch to `Console` tab
2. Refresh the page
3. Start download, wait until you get that `Wrong MIME` error
4. Copy and paste all the logs here

By the way, what's your browser and userscript extension?

**Comment by bluefiberbread** [comment]
Firefox 56.0.2

https://puu.sh/z53qS/c5b5d70631.jpg
https://puu.sh/z53rm/bdcc6d0f0f.jpg

**Comment by ccloli** [comment]
Hi, I checked your screenshot and the gallery you downloaded, seems that the other images that have the original image link didn't have the problem, also looked like the failed one's HTTP header was from E-Hentai itself. So...maybe it's the bug of E-Hentai server. If you can get the image when the script retry automatically, you can ignore this `Wrong MIME` error.

**Comment by bluefiberbread** [comment]
yeah  I think it's their servers too. but you said you could download it just fine, what galleries did you download? I'll try the same

**Comment by bluefiberbread** [comment]
is it normal for the script to ask me to remind me that I'm not online on the forum when I am, I've logged out and in and it's giving me the same error.

**Comment by ccloli** [comment]
Just [the gallery](https://exhentai.org/g/1167700/8a6450bb7c) on your screenshot, I didn't get your `Wrong MIME` error, only `Network Error` when it's downloading due to my unstable network. 

The script checks your login status by checking a value on the site, you can execute `apiuid` at F12 -> Console, the returned value is your forum uid. It should be grater than 0, and if it's `-1`, meaning that you are not login. However you can access to ExHentai, that means you're already logged in.

A possible reason is that your cookies is not sync with forum. #86 says that he can visit ExHentai but cannot download, and it's due to wrong cookie setting. Please consider to:
1. Clear your cookies of E-Hentai and ExHentai
    a. Go to `about:preferences#privacy`
    b. Click that *remove individual cookies* link at **History** section
    c. Search `e-hentai.org` and  `exhentai.org`
    d. Click the _Remove All Shown_ button
2. Check your browser settings that you've enabled **Accept cookies from websites**, **Accept third-party cookies** 
    a. Go to `about:preferences#privacy`
    b. Set _Firefox will ..._ to _Remember history_ *
3. Go to https://e-hentai.org/home.php to login

\* If you prefer to use _Use custom settings for history_ for your other websites security, follow the steps below
1. Enable _Accept cookies from websites_
2. Click the _Exceptions..._ button
3. Input `https://e-hentai.org` and `https://exhentai.org`, click the _Allow_ button

**Comment by bluefiberbread** [comment]
Yup it works now, thanks for the help :)

**Comment by maximovich** [comment]
i did everything in this post, when i download gallery popup say 
"You are not logged in to E-Hentai Forums, so you can't download original image. Continue?"
url:
https://exhentai.org/g/841674/cf73369854/
console:
[EHD] Is Logged In > false
but i am logged in e-hentai forum

**Comment by ccloli** [comment]
@maximovich Please try clearing your Cookies, and try again. BTW, what's your browser and script manager? Looks like the reports about cookies is increased these days.

**Comment by maximovich** [comment]
Sorry for bothering you, was my mistake! in firefox v60.0.2 to remove cookies it's not enough with click the Remove button, you must also click the Save Changes button. 
Your programing skill is really apreciated.

**Comment by 70338** [comment]
I have some problem
I canot download gallery because of the wrong mime
Can you slove it ,please
https://imgur.com/Zf0VYQm

**Comment by ccloli** [comment]
> I have some problem
> I canot download gallery because of the wrong mime
> Can you slove it ,please
> https://imgur.com/Zf0VYQm

@70338 See #153 for detail. TL;DR: Update to v1.31.6.

**Comment by 70338** [comment]
thank you 
i found the update 
love your javascript
thankyou
